### PR TITLE
Feature/view all vendor

### PR DESCRIPTION
**Pull request by valenciusap17**
## Describe your changes
- Create test for get all vendor accessor
- Create test for get all vendor service
- Add get all vendor accessor implementation
- Add get all vendor service implementation

## Issue ticket number and link
- ADU-23: https://linear.app/adudu/issue/ADU-23/ut-for-get-list-of-vendors-api
- ADU-16: https://linear.app/adudu/issue/ADU-16/get-list-of-vendors-api

### CODE REVIEW

File: internal/vendors/accessor_test.go [hunk 1]
```diff
@@ -6,15 +6,12 @@ import (
...
+	"time"
 
 	"github.com/DATA-DOG/go-sqlmock"
 	"github.com/onsi/gomega"
 )
 
-func Test_newPostgresVendorAccessor(t *testing.T) {
-	_ = newPostgresVendorAccessor(nil)
-}
-
```
Comment (valeelim) [inline internal/vendors/accessor_test.go:17]: keep this, for coverage and ensuring the function doesn't panic

File: internal/vendors/accessor_test.go [hunk 2]
```diff
@@ -6,15 +6,12 @@ import (
 	"errors"
 	"log"
 	"testing"
+	"time"
 
 	"github.com/DATA-DOG/go-sqlmock"
 	"github.com/onsi/gomega"
 )
 
-func Test_newPostgresVendorAccessor(t *testing.T) {
```
Comment (edutjie) [inline internal/vendors/accessor_test.go:14]: why is this deleted?

File: internal/vendors/service.go [hunk 3]
```diff
@@ -15,6 +16,14 @@ func (v *VendorService) GetSomeStuff(ctx context.Context) ([]string, error) {
 	return v.vendorDBAccessor.GetSomeStuff(ctx)
 }
 
+func (v *VendorService) GetAll(ctx context.Context) ([]Vendor, error) {
+	vendors, err := v.vendorDBAccessor.GetAll(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return vendors, nil
```
Comment (valeelim) [inline internal/vendors/service.go:24]: lgsg `return v.vendorDBAccessor.GetAll(ctx)` aja

File: internal/vendors/service_test.go [hunk 4]
```diff
@@ -52,3 +55,67 @@ func TestVendorService_GetSomeStuff(t *testing.T) {
...
+			vendors, err := v.GetAll(tt.args.ctx)
+
+			if (err != nil) != tt.wantErr {
+				t.Errorf("GetAll() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			if !reflect.DeepEqual(vendors, tt.want) {
+				t.Errorf("GetAll() = %v, wantErr %v", vendors, tt.wantErr)
+			}
```
Comment (valeelim) [inline internal/vendors/service_test.go:118]: bisa pake `gomega` aj ga, spy seragam

File: internal/vendors/accessor_test.go [hunk 5]
```diff
@@ -87,3 +84,191 @@ func Test_postgresVendorAccessor_GetSomeStuff(t *testing.T) {
...
+
+		g.Expect(err).To(gomega.BeNil())
+		g.Expect(res).To(gomega.Equal(expectation))
+	})
+
+	t.Run("success on empty result", func(t *testing.T) {
+		g, db := setup(t)
+		defer db.Close()
+
+		sampleData := []string{
```
Comment (edutjie) [inline internal/vendors/accessor_test.go:184]: I think this is duplicated (?) why not just use the sampleData defined before?

File: internal/vendors/accessor_test.go [hunk 6]
```diff
@@ -87,3 +84,191 @@ func Test_postgresVendorAccessor_GetSomeStuff(t *testing.T) {
...
+		res, err := accessor.GetAll(ctx)
+		g.Expect(err).To(gomega.BeNil())
+		g.Expect(res).To(gomega.Equal([]Vendor{}))
+	})
+
+	t.Run("error on scanning row", func(t *testing.T) {
+		g, db := setup(t)
+		defer db.Close()
+
+		sampleData := []string{
```
Comment (edutjie) [inline internal/vendors/accessor_test.go:225]: ditto

File: internal/vendors/accessor_test.go [hunk 7]
```diff
@@ -87,3 +84,191 @@ func Test_postgresVendorAccessor_GetSomeStuff(t *testing.T) {
...
+			"name",
+			"bp_id",
+			"bp_name",
+			"rating",
+			"area_group_id",
+			"area_group_name",
+			"sap_code",
+			"modified_date",
+			"modified_by",
+			"dt",
```
Comment (valeelim) [inline internal/vendors/accessor_test.go:195]: ini ga perlu di declare ulang, di scope atas udh ada

File: internal/vendors/accessor_test.go [hunk 8]
```diff
@@ -87,3 +84,191 @@ func Test_postgresVendorAccessor_GetSomeStuff(t *testing.T) {
...
+			"bp_id",
+			"bp_name",
+			"rating",
+			"area_group_id",
+			"area_group_name",
+			"sap_code",
+			"modified_date",
+			"modified_by",
+			"dt" 
+			FROM vendor`).
```
Comment (valeelim) [inline internal/vendors/accessor_test.go:156]: querynya dibikin jd var aj spy ga repetitive

File: internal/vendors/accessor_test.go (hunk 6 above)
Comment (valeelim) [inline internal/vendors/accessor_test.go:225]: ini juga

File: internal/vendors/service_test.go [hunk 9]
```diff
@@ -52,3 +55,67 @@ func TestVendorService_GetSomeStuff(t *testing.T) {
...
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    []Vendor
+		wantErr bool
+	}{
+		// TODO: Add test cases.
+		{
+			name: "Test_GetAllVendor Success",
```
Comment (valeelim) [inline internal/vendors/service_test.go:92]: tambah test false

File: internal/vendors/vendor.go [hunk 10]
```diff
@@ -1,7 +1,19 @@
 package vendors
 
+import "time"
+
 // Vendor defines the metadata related to a vendor
 // i.e. name, etc
 type Vendor struct {
-	// TO-DO
+	Id            int
```
Comment (valeelim) [inline internal/vendors/vendor.go:8]: - ID pake string aj, spy lebih flexible (cth tambah prefix)
- Tambah struct tag ke masing" field, supaya bs extract dr fixtures json, i.e. `json:"id"`

**Comment by ferryhieronimus** [comment]
Others have commented on your code. I think you should describe your PR. Should we agree on a standard format for PR descriptions?

**Comment by valeelim** [comment]
> Others have commented on your code. I think you should describe your PR. Should we agree on a standard format for PR descriptions?

That's a great idea, i think we could use https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository as a reference

**Comment by ferryhieronimus** [comment]
> > Others have commented on your code. I think you should describe your PR. Should we agree on a standard format for PR descriptions?
> 
> That's a great idea, i think we could use https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository as a reference

TIL you can add PR template on GitHub. Do you have any suggestions on the message itself?

File: internal/vendors/accessor_test.go (hunk 1 above)
Comment (valenciusap17) [reply internal/vendors/accessor_test.go:17]: Ahh okay, unintentionally deleted while i was resolving conflicts My bad ✌️

File: internal/vendors/service_test.go (hunk 4 above)
Comment (valenciusap17) [reply internal/vendors/service_test.go:118]: Ohh benar juga, okeh2

**Comment by valenciusap17** [comment]
> > > Others have commented on your code. I think you should describe your PR. Should we agree on a standard format for PR descriptions?
> > 
> > 
> > That's a great idea, i think we could use https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository as a reference
> 
> TIL you can add PR template on GitHub. Do you have any suggestions on the message itself?

How about the example 1 from https://axolo.co/blog/p/part-3-github-pull-request-template?

File: internal/vendors/service_test.go (hunk 9 above)
Comment (valenciusap17) [reply internal/vendors/service_test.go:92]: Fail test is just changing the "want" value and "wantErr" to true ya?  Not yet familiar with doing test ehehehe

**Comment by ferryhieronimus** [comment]
> What about the example 1 from https://axolo.co/blog/p/part-3-github-pull-request-template?

Looks great and simple, but i dont think we need the "Checklist before requesting a review" part

**Comment by valenciusap17** [comment]
> Looks great and simple, but i dont think we need the "Checklist before requesting a review" part

Yap agree, I will try to create the pull request template according to the reference mentioned ya

File: internal/vendors/accessor_test.go [hunk 11]
```diff
@@ -87,3 +88,141 @@ func Test_postgresVendorAccessor_GetSomeStuff(t *testing.T) {
...
+			"bp_id",
+			"bp_name",
+			"rating",
+			"area_group_id",
+			"area_group_name",
+			"sap_code",
+			"modified_date",
+			"modified_by",
+			"dt" 
+			FROM vendor`
```
Comment (edutjie) [inline internal/vendors/accessor_test.go:139]: I think you should also return description

Comment (valenciusap17) [reply internal/vendors/accessor_test.go:141]: Yea, sorry I missed it. Thanks for the double check bro 👍

**Review by edutjie — APPROVED** [review]
LGTM

File: internal/vendors/service_test.go [hunk 12]
```diff
@@ -52,3 +56,71 @@ func TestVendorService_GetSomeStuff(t *testing.T) {
...
+			wantErr: false,
+		},
+		{
+			name: "failure",
+			fields: fields{
+				mockVendorDBAccessor: NewMockvendorDBAccessor(ctrl),
+			},
+			args:    args{ctx: context.Background()},
+			want:    nil,
+			wantErr: true,
```
Comment (edutjie) [inline internal/vendors/service_test.go:108]: Sorry, I just noticed. Is `wantErr` expected not to be used? I though it should we used below in the for loop to throw some error if it's true (?) 

cc: @valeelim

**Review by valeelim — CHANGES_REQUESTED** [review]

**Comment by valeelim** [comment]
LGTM ❤️

**Review by valeelim — APPROVED** [review]
LGTM
